src/nvme/util.c

```diff
@@ -419,6 +423,38 @@ void nvme_init_copy_range_f1(struct nvme_copy_range_f1 *copy, __u16 *nlbs,
 	}  
 }
 
+void nvme_init_copy_range_f2(struct nvme_copy_range_f2 *copy, __u32 *snsids,
+			  __u16 *nlbs, __u64 *slbas, __u16 *sopts,
+			  __u32 *eilbrts, __u32 *elbatms, __u32 *elbats,
+			  __u16 nr)
+{
+	int i;
+
+	nvme_init_copy_range((struct nvme_copy_range *)copy, nlbs, slbas,
```

Comment:
It is undefined behavior to access storage through a pointer type that doesn't match the type the storage was declared with. (This is because the two types may have different alignment requirements, and because C compilers are allowed to assume pointers of different types don't alias.)

There are a few ways to avoid this. One is to just repeat the field assignments in each of the `nvme_init_copy_range_f*()` functions. Another is to define a `struct nvme_copy_range` on the stack, pass a pointer to it into `nvme_init_copy_range()`, and then `memcpy()` it to the `nvme_copy_range_f2`. Or a macro to deduplicate the common code in these functions. And there are plenty of other possible ways to do this...